<a href="https://colab.research.google.com/github/dsdayat/Assignment-DS-Day-7-Dibimbing/blob/main/HW_HTnc_Agus_Nur_Hidayat.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**Hypothesis testing**

Tugas nya adalah lakukan pengujian hipotesis untuk
membuktikan apakah education berpengaruh terhadap
income atau tidak?



In [232]:
import pandas as pd
import numpy as np
import scipy.stats as st
import seaborn as sns

In [71]:
df=pd.read_csv('marketing_campaign.csv',sep="\t")

In [72]:
df.shape


(2240, 29)

In [141]:
#cek missing value kalo ada, nembak pake mean karena aku gatau harus diapain.
df.isnull().sum() 

ID                      0
Year_Birth              0
Education               0
Marital_Status          0
Income                 24
Kidhome                 0
Teenhome                0
Dt_Customer             0
Recency                 0
MntWines                0
MntFruits               0
MntMeatProducts         0
MntFishProducts         0
MntSweetProducts        0
MntGoldProds            0
NumDealsPurchases       0
NumWebPurchases         0
NumCatalogPurchases     0
NumStorePurchases       0
NumWebVisitsMonth       0
AcceptedCmp3            0
AcceptedCmp4            0
AcceptedCmp5            0
AcceptedCmp1            0
AcceptedCmp2            0
Complain                0
Z_CostContact           0
Z_Revenue               0
Response                0
dtype: int64

In [146]:
nembakmean=df['Income'].mean()
df['Income']=df['Income'].fillna(nembakmean)
df.isnull().sum() 


ID                     0
Year_Birth             0
Education              0
Marital_Status         0
Income                 0
Kidhome                0
Teenhome               0
Dt_Customer            0
Recency                0
MntWines               0
MntFruits              0
MntMeatProducts        0
MntFishProducts        0
MntSweetProducts       0
MntGoldProds           0
NumDealsPurchases      0
NumWebPurchases        0
NumCatalogPurchases    0
NumStorePurchases      0
NumWebVisitsMonth      0
AcceptedCmp3           0
AcceptedCmp4           0
AcceptedCmp5           0
AcceptedCmp1           0
AcceptedCmp2           0
Complain               0
Z_CostContact          0
Z_Revenue              0
Response               0
dtype: int64

In [154]:
df.head()

ID  Year_Birth   Education Marital_Status   Income  Kidhome  Teenhome  \
0  5524        1957  Graduation         Single  58138.0        0         0   
1  2174        1954  Graduation         Single  46344.0        1         1   
2  4141        1965  Graduation       Together  71613.0        0         0   
3  6182        1984  Graduation       Together  26646.0        1         0   
4  5324        1981         PhD        Married  58293.0        1         0   

  Dt_Customer  Recency  MntWines  ...  NumWebVisitsMonth  AcceptedCmp3  \
0  04-09-2012       58       635  ...                  7             0   
1  08-03-2014       38        11  ...                  5             0   
2  21-08-2013       26       426  ...                  4             0   
3  10-02-2014       26        11  ...                  6             0   
4  19-01-2014       94       173  ...                  5             0   

   AcceptedCmp4  AcceptedCmp5  AcceptedCmp1  AcceptedCmp2  Complain  \
0             0             0             0             0         0   
1             0             0             0             0         0   
2             0             0             0             0         0   
3             0             0             0             0         0   
4             0             0             0             0         0   

   Z_CostContact  Z_Revenue  Response  
0              3         11         1  
1              3         11         0  
2              3         11         0  
3              3         11         0  
4              3         11         0  

[5 rows x 29 columns]

In [148]:
df.tail()

ID  Year_Birth   Education Marital_Status   Income  Kidhome  \
2235  10870        1967  Graduation        Married  61223.0        0   
2236   4001        1946         PhD       Together  64014.0        2   
2237   7270        1981  Graduation       Divorced  56981.0        0   
2238   8235        1956      Master       Together  69245.0        0   
2239   9405        1954         PhD        Married  52869.0        1   

      Teenhome Dt_Customer  Recency  MntWines  ...  NumWebVisitsMonth  \
2235         1  13-06-2013       46       709  ...                  5   
2236         1  10-06-2014       56       406  ...                  7   
2237         0  25-01-2014       91       908  ...                  6   
2238         1  24-01-2014        8       428  ...                  3   
2239         1  15-10-2012       40        84  ...                  7   

      AcceptedCmp3  AcceptedCmp4  AcceptedCmp5  AcceptedCmp1  AcceptedCmp2  \
2235             0             0             0             0             0   
2236             0             0             0             1             0   
2237             0             1             0             0             0   
2238             0             0             0             0             0   
2239             0             0             0             0             0   

      Complain  Z_CostContact  Z_Revenue  Response  
2235         0              3         11         0  
2236         0              3         11         0  
2237         0              3         11         0  
2238         0              3         11         0  
2239         0              3         11         1  

[5 rows x 29 columns]

In [149]:
N = 2240 # jumlah populasinya
e = 0.05 # parameter margin of errornya

In [150]:
def slovin(N,e):
    return N/(1+N * e**2)

In [213]:
Ns = slovin(N,e) 
print(Ns)

339.3939393939393


**Sample proportional stratified**

Populasi =2240 Sample dengan sample slovin = 339  . karena kita mau pake proportionate stratified random sampling jadi kita cari tahu jumlah populasi di tiap-tiap kelas variabel bebasnya dengan "value_counts()" atau "df.groupby(['kolom'],as_index=False).count()"
CMIIW



In [152]:
df['Education'].value_counts()

Graduation    1127
PhD            486
Master         370
2n Cycle       203
Basic           54
Name: Education, dtype: int64

In [174]:
df.groupby(['Education'],as_index=False).count()

Education    ID  Year_Birth  Marital_Status  Income  Kidhome  Teenhome  \
0    2n Cycle   203         203             203     203      203       203   
1       Basic    54          54              54      54       54        54   
2  Graduation  1127        1127            1127    1127     1127      1127   
3      Master   370         370             370     370      370       370   
4         PhD   486         486             486     486      486       486   

   Dt_Customer  Recency  MntWines  ...  NumWebVisitsMonth  AcceptedCmp3  \
0          203      203       203  ...                203           203   
1           54       54        54  ...                 54            54   
2         1127     1127      1127  ...               1127          1127   
3          370      370       370  ...                370           370   
4          486      486       486  ...                486           486   

   AcceptedCmp4  AcceptedCmp5  AcceptedCmp1  AcceptedCmp2  Complain  \
0           203           203           203           203       203   
1            54            54            54            54        54   
2          1127          1127          1127          1127      1127   
3           370           370           370           370       370   
4           486           486           486           486       486   

   Z_CostContact  Z_Revenue  Response  
0            203        203       203  
1             54         54        54  
2           1127       1127      1127  
3            370        370       370  
4            486        486       486  

[5 rows x 29 columns]

In [182]:
p0=203
p1=54
p2=1127
p3=370
p4=486

In [216]:
#mencari Npengali tiap kelas (Np)
Np = Ns/N
print(Np)

0.1515151515151515


In [218]:
#menentukan jumlah sample tiap kelas
Np0=Np*p0
print ("ntk 0=  ", Np0)
Np1=Np*p1
print ("ntk 1=  ", Np1)
Np2=Np*p2
print ("ntk 2=  ", Np2)
Np3=Np*p3
print ("ntk 3=  ", Np3)
Np4=Np*p4
print ("ntk 4=  ", Np4)

ntk 0=   30.757575757575754
ntk 1=   8.18181818181818
ntk 2=   170.75757575757572
ntk 3=   56.060606060606055
ntk 4=   73.63636363636363


Karena terdapat 5 kelas tingkat pendidikan pada variabel education dengan populasi yang berbeda maka pengambilan sampel dilakukan dengan hasil slovin dibagi hasil kelas 339/5 sebagai Np alias Npembagi. 

Besarnya sampel tiap kelas (Ntk) = Np/p*Ns

dan dapet

ntk 0=   30.757575757575754==> 31

ntk 1=   8.18181818181818==>8

ntk 2=   170.75757575757572==>171

ntk 3=   56.060606060606055==>56

ntk 4=   73.63636363636363==>74



In [219]:
#mendefine kelas education berdasarkan index dan mendklarasi jumlah sampelnya
df0 = df[df['Education'] == '2n Cycle']
df1 = df[df['Education'] == 'Basic']
df2 = df[df['Education'] == 'Graduation']
df3 = df[df['Education'] == 'Master']
df4 = df[df['Education'] == 'PhD']

df0c=df0.sample(31)
df1c=df1.sample(8)
df2c=df2.sample(171)
df3c=df3.sample(56)
df4c=df4.sample(74)


In [220]:
#gabungkan sample tiap kelas ke kedalam sample education, harus sama dengan nilai slovin.
Nedu =pd.concat([df0c, df1c,df2c,df3c,df4c],axis=0)


In [240]:
print(Nedu['Education'])

1097    2n Cycle
239     2n Cycle
78      2n Cycle
370     2n Cycle
991     2n Cycle
          ...   
1025         PhD
1526         PhD
1577         PhD
828          PhD
679          PhD
Name: Education, Length: 340, dtype: object


In [223]:
Nedu['Education'].value_counts()#ini detail samplenya

Graduation    171
PhD            74
Master         56
2n Cycle       31
Basic           8
Name: Education, dtype: int64

In [224]:
Nedu.shape#ini total sample sudah didapatkan Nedu, ingat harus sama dengan nilai sample slovin

(340, 29)

**Lanjut ke Anova test**

Karena variabel bebasnya banyak (tipe stratified) makanya pake anova.

Dan karena sudah dapet sample yang dipisah bisa langsung uji anova.

digunakan kriiteria alpha =0.05



H0:df0=df1=df2=df3=df4  (Tidak ada mean yang berbeda dalam kelompok kategori)

H1 = Sedikitnya ada satu mean yang berbeda dalam kelompok kategori

F1<pv = education tidak berpengaruh terhadap income

F1>pv = education berpengaruh terhadap income

In [241]:
alp=0.05
anova_test = st.f_oneway(df0c['Income'],
                         df1c['Income'],
                         df2c['Income'],
                         df3c['Income'],
                         df4c['Income'],
                         )

In [234]:
pv= anova_test.pvalue
print(pv)

0.2038319719677809


In [231]:
if (alp<e):
  print("Education tidak berpengaruh terhadap income")
else:
  print("Education berpengaruh terhadap income")

Education berpengaruh terhadap income


**Statistika Deskriptif**



In [249]:
df0c.describe()

ID   Year_Birth        Income    Kidhome   Teenhome  \
count     31.000000    31.000000     31.000000  31.000000  31.000000   
mean    5308.032258  1973.483871  51195.677419   0.580645   0.387097   
std     3156.964771    18.156488  20700.633637   0.564163   0.495138   
min       89.000000  1893.000000  22212.000000   0.000000   0.000000   
25%     2777.500000  1967.000000  35536.000000   0.000000   0.000000   
50%     4279.000000  1976.000000  46831.000000   1.000000   0.000000   
75%     8194.000000  1984.000000  70876.500000   1.000000   1.000000   
max    11004.000000  1995.000000  85485.000000   2.000000   1.000000   

         Recency    MntWines   MntFruits  MntMeatProducts  MntFishProducts  \
count  31.000000   31.000000   31.000000        31.000000        31.000000   
mean   55.225806  162.064516   32.290323       137.806452        36.096774   
std    28.326324  244.752519   46.221347       215.832253        46.908674   
min     1.000000    1.000000    0.000000         2.000000         0.000000   
25%    31.000000    9.000000    3.500000         7.500000         3.000000   
50%    56.000000   46.000000   12.000000        37.000000         7.000000   
75%    81.500000  164.000000   35.500000       118.000000        56.500000   
max    99.000000  938.000000  193.000000       754.000000       159.000000   

       ...  NumWebVisitsMonth  AcceptedCmp3  AcceptedCmp4  AcceptedCmp5  \
count  ...          31.000000     31.000000     31.000000     31.000000   
mean   ...           4.677419      0.064516      0.032258      0.096774   
std    ...           2.357783      0.249731      0.179605      0.300537   
min    ...           1.000000      0.000000      0.000000      0.000000   
25%    ...           3.000000      0.000000      0.000000      0.000000   
50%    ...           4.000000      0.000000      0.000000      0.000000   
75%    ...           7.000000      0.000000      0.000000      0.000000   
max    ...           8.000000      1.000000      1.000000      1.000000   

       AcceptedCmp1  AcceptedCmp2   Complain  Z_CostContact  Z_Revenue  \
count     31.000000          31.0  31.000000           31.0       31.0   
mean       0.064516           0.0   0.064516            3.0       11.0   
std        0.249731           0.0   0.249731            0.0        0.0   
min        0.000000           0.0   0.000000            3.0       11.0   
25%        0.000000           0.0   0.000000            3.0       11.0   
50%        0.000000           0.0   0.000000            3.0       11.0   
75%        0.000000           0.0   0.000000            3.0       11.0   
max        1.000000           0.0   1.000000            3.0       11.0   

        Response  
count  31.000000  
mean    0.096774  
std     0.300537  
min     0.000000  
25%     0.000000  
50%     0.000000  
75%     0.000000  
max     1.000000  

[8 rows x 26 columns]

In [250]:
df1c.describe()

ID  Year_Birth        Income   Kidhome  Teenhome    Recency  \
count      8.00000     8.00000      8.000000  8.000000   8.00000   8.000000   
mean    5076.12500  1974.87500  22417.875000  0.375000   0.25000  40.875000   
std     3179.79579    13.66369   5587.895015  0.517549   0.46291  19.995982   
min     1744.00000  1947.00000  14188.000000  0.000000   0.00000   5.000000   
25%     2339.50000  1970.25000  19464.000000  0.000000   0.00000  31.250000   
50%     4696.00000  1978.50000  22296.000000  0.000000   0.00000  44.500000   
75%     6595.75000  1985.25000  25372.500000  1.000000   0.25000  52.750000   
max    10250.00000  1987.00000  30801.000000  1.000000   1.00000  67.000000   

       MntWines  MntFruits  MntMeatProducts  MntFishProducts  ...  \
count  8.000000   8.000000         8.000000         8.000000  ...   
mean   3.500000   6.500000         6.750000         8.125000  ...   
std    3.251373   3.162278         4.234214         7.414994  ...   
min    1.000000   3.000000         2.000000         0.000000  ...   
25%    1.000000   4.000000         3.750000         2.750000  ...   
50%    2.000000   6.000000         5.000000         5.000000  ...   
75%    5.000000   7.750000        11.000000        15.250000  ...   
max    9.000000  12.000000        13.000000        19.000000  ...   

       NumWebVisitsMonth  AcceptedCmp3  AcceptedCmp4  AcceptedCmp5  \
count           8.000000           8.0           8.0           8.0   
mean            6.500000           0.0           0.0           0.0   
std             1.414214           0.0           0.0           0.0   
min             4.000000           0.0           0.0           0.0   
25%             6.000000           0.0           0.0           0.0   
50%             6.500000           0.0           0.0           0.0   
75%             7.000000           0.0           0.0           0.0   
max             9.000000           0.0           0.0           0.0   

       AcceptedCmp1  AcceptedCmp2  Complain  Z_CostContact  Z_Revenue  \
count           8.0           8.0       8.0            8.0        8.0   
mean            0.0           0.0       0.0            3.0       11.0   
std             0.0           0.0       0.0            0.0        0.0   
min             0.0           0.0       0.0            3.0       11.0   
25%             0.0           0.0       0.0            3.0       11.0   
50%             0.0           0.0       0.0            3.0       11.0   
75%             0.0           0.0       0.0            3.0       11.0   
max             0.0           0.0       0.0            3.0       11.0   

       Response  
count       8.0  
mean        0.0  
std         0.0  
min         0.0  
25%         0.0  
50%         0.0  
75%         0.0  
max         0.0  

[8 rows x 26 columns]

In [251]:
df2c.describe()

ID   Year_Birth         Income     Kidhome    Teenhome  \
count    171.000000   171.000000     171.000000  171.000000  171.000000   
mean    6075.584795  1968.116959   56751.336273    0.409357    0.573099   
std     3168.574335    11.168143   51242.476309    0.549572    0.552195   
min       78.000000  1944.000000    1730.000000    0.000000    0.000000   
25%     3424.500000  1959.000000   37414.000000    0.000000    0.000000   
50%     6200.000000  1969.000000   53201.000000    0.000000    1.000000   
75%     9147.000000  1977.000000   70426.000000    1.000000    1.000000   
max    11100.000000  1993.000000  666666.000000    2.000000    2.000000   

          Recency     MntWines   MntFruits  MntMeatProducts  MntFishProducts  \
count  171.000000   171.000000  171.000000       171.000000       171.000000   
mean    51.543860   297.011696   29.111111       165.444444        44.204678   
std     28.321725   325.362133   41.637433       209.398245        59.888319   
min      1.000000     1.000000    0.000000         1.000000         0.000000   
25%     28.000000    34.500000    2.000000        15.000000         3.000000   
50%     53.000000   199.000000    8.000000        84.000000        17.000000   
75%     73.500000   457.000000   37.000000       223.500000        56.500000   
max     99.000000  1324.000000  189.000000       936.000000       240.000000   

       ...  NumWebVisitsMonth  AcceptedCmp3  AcceptedCmp4  AcceptedCmp5  \
count  ...          171.00000    171.000000    171.000000    171.000000   
mean   ...            5.19883      0.070175      0.087719      0.076023   
std    ...            2.68464      0.256193      0.283717      0.265814   
min    ...            1.00000      0.000000      0.000000      0.000000   
25%    ...            3.00000      0.000000      0.000000      0.000000   
50%    ...            5.00000      0.000000      0.000000      0.000000   
75%    ...            7.00000      0.000000      0.000000      0.000000   
max    ...           20.00000      1.000000      1.000000      1.000000   

       AcceptedCmp1  AcceptedCmp2  Complain  Z_CostContact  Z_Revenue  \
count    171.000000    171.000000     171.0          171.0      171.0   
mean       0.076023      0.023392       0.0            3.0       11.0   
std        0.265814      0.151588       0.0            0.0        0.0   
min        0.000000      0.000000       0.0            3.0       11.0   
25%        0.000000      0.000000       0.0            3.0       11.0   
50%        0.000000      0.000000       0.0            3.0       11.0   
75%        0.000000      0.000000       0.0            3.0       11.0   
max        1.000000      1.000000       0.0            3.0       11.0   

         Response  
count  171.000000  
mean     0.152047  
std      0.360121  
min      0.000000  
25%      0.000000  
50%      0.000000  
75%      0.000000  
max      1.000000  

[8 rows x 26 columns]

In [252]:
df3c.describe()

ID   Year_Birth         Income    Kidhome   Teenhome  \
count     56.000000    56.000000      56.000000  56.000000  56.000000   
mean    5755.857143  1966.553571   56407.772346   0.446429   0.517857   
std     3027.349044    11.833573   24100.721129   0.501621   0.571794   
min      203.000000  1943.000000   16653.000000   0.000000   0.000000   
25%     3540.750000  1956.000000   40114.750000   0.000000   0.000000   
50%     5394.500000  1968.000000   54539.000000   0.000000   0.000000   
75%     7949.000000  1975.000000   74146.000000   1.000000   1.000000   
max    10991.000000  1992.000000  157733.000000   1.000000   2.000000   

         Recency     MntWines   MntFruits  MntMeatProducts  MntFishProducts  \
count  56.000000    56.000000   56.000000        56.000000        56.000000   
mean   41.625000   365.714286   25.642857       173.946429        30.285714   
std    29.256584   374.844875   42.190662       228.305730        53.877880   
min     0.000000     2.000000    0.000000         2.000000         0.000000   
25%    15.500000    45.000000    1.750000        17.750000         2.000000   
50%    39.000000   214.500000    6.000000        67.000000         6.000000   
75%    66.000000   547.250000   36.250000       247.750000        28.000000   
max    94.000000  1288.000000  194.000000       921.000000       253.000000   

       ...  NumWebVisitsMonth  AcceptedCmp3  AcceptedCmp4  AcceptedCmp5  \
count  ...          56.000000     56.000000     56.000000     56.000000   
mean   ...           4.928571      0.089286      0.035714      0.071429   
std    ...           2.295111      0.287736      0.187256      0.259870   
min    ...           1.000000      0.000000      0.000000      0.000000   
25%    ...           3.000000      0.000000      0.000000      0.000000   
50%    ...           5.000000      0.000000      0.000000      0.000000   
75%    ...           7.000000      0.000000      0.000000      0.000000   
max    ...           9.000000      1.000000      1.000000      1.000000   

       AcceptedCmp1  AcceptedCmp2  Complain  Z_CostContact  Z_Revenue  \
count     56.000000          56.0      56.0           56.0       56.0   
mean       0.107143           0.0       0.0            3.0       11.0   
std        0.312094           0.0       0.0            0.0        0.0   
min        0.000000           0.0       0.0            3.0       11.0   
25%        0.000000           0.0       0.0            3.0       11.0   
50%        0.000000           0.0       0.0            3.0       11.0   
75%        0.000000           0.0       0.0            3.0       11.0   
max        1.000000           0.0       0.0            3.0       11.0   

        Response  
count  56.000000  
mean    0.160714  
std     0.370591  
min     0.000000  
25%     0.000000  
50%     0.000000  
75%     0.000000  
max     1.000000  

[8 rows x 26 columns]

In [253]:
df4c.describe()

ID   Year_Birth         Income    Kidhome   Teenhome  \
count     74.000000    74.000000      74.000000  74.000000  74.000000   
mean    5409.608108  1966.378378   56126.628415   0.418919   0.554054   
std     3437.148288    12.514871   26133.689580   0.523601   0.552501   
min       73.000000  1944.000000   14849.000000   0.000000   0.000000   
25%     2123.000000  1958.000000   37480.000000   0.000000   0.000000   
50%     5143.000000  1968.000000   54741.000000   0.000000   1.000000   
75%     8373.750000  1975.500000   68123.750000   1.000000   1.000000   
max    11010.000000  1991.000000  162397.000000   2.000000   2.000000   

         Recency     MntWines   MntFruits  MntMeatProducts  MntFishProducts  \
count  74.000000    74.000000   74.000000        74.000000        74.000000   
mean   52.054054   345.945946   18.959459       167.851351        20.756757   
std    29.351386   398.710452   35.669502       256.682560        33.417578   
min     6.000000     6.000000    0.000000         3.000000         0.000000   
25%    22.250000    22.250000    0.000000        18.000000         2.000000   
50%    55.500000   178.000000    3.000000        38.000000         5.000000   
75%    78.750000   595.000000   19.750000       229.500000        23.000000   
max    98.000000  1332.000000  194.000000      1582.000000       149.000000   

       ...  NumWebVisitsMonth  AcceptedCmp3  AcceptedCmp4  AcceptedCmp5  \
count  ...          74.000000     74.000000     74.000000     74.000000   
mean   ...           5.189189      0.027027      0.054054      0.054054   
std    ...           2.557159      0.163269      0.227668      0.227668   
min    ...           0.000000      0.000000      0.000000      0.000000   
25%    ...           3.000000      0.000000      0.000000      0.000000   
50%    ...           5.000000      0.000000      0.000000      0.000000   
75%    ...           7.750000      0.000000      0.000000      0.000000   
max    ...           9.000000      1.000000      1.000000      1.000000   

       AcceptedCmp1  AcceptedCmp2  Complain  Z_CostContact  Z_Revenue  \
count     74.000000          74.0      74.0           74.0       74.0   
mean       0.067568           0.0       0.0            3.0       11.0   
std        0.252716           0.0       0.0            0.0        0.0   
min        0.000000           0.0       0.0            3.0       11.0   
25%        0.000000           0.0       0.0            3.0       11.0   
50%        0.000000           0.0       0.0            3.0       11.0   
75%        0.000000           0.0       0.0            3.0       11.0   
max        1.000000           0.0       0.0            3.0       11.0   

        Response  
count  74.000000  
mean    0.189189  
std     0.394332  
min     0.000000  
25%     0.000000  
50%     0.000000  
75%     0.000000  
max     1.000000  

[8 rows x 26 columns]